wallet/src/routes.rs

```diff
@@ -301,6 +304,36 @@ pub fn wallet_error(source: CapeWalletError) -> tide::Error {
     })
 }
 
+pub fn get_home_path() -> Result<PathBuf, tide::Error> {
+    let home = std::env::var("HOME").map_err(|_| {
+        server_error(CapeAPIError::Internal {
+            msg: String::from(
+                "HOME directory is not set. Please set the server's HOME directory, or \
+                    specify a different storage location using :path.",
+            ),
+        })
+    })?;
+    Ok(PathBuf::from(home))
+}
+
+pub async fn write_path(wallet_path: &Path) -> Result<(), tide::Error> {
+    let mut storage_path = get_home_path().unwrap();
+    storage_path.push(".espresso/cape");
+    create_dir_all(&storage_path).await?;
+    storage_path.push("last_wallet_path");
+    let mut file = File::create(storage_path).await?;
+    Ok(file
+        .write_all(&bincode::serialize(&wallet_path).unwrap())
+        .await?)
+}
+pub async fn read_last_path() -> Result<PathBuf, tide::Error> {
+    let mut path = get_home_path().unwrap();
+    path.push(".espresso/cape/last_wallet_path");
+    let mut file = File::open(&path).await?;
+    let mut bytes = Vec::new();
+    file.read_to_end(&mut bytes).await?;
+    Ok(bincode::deserialize(&bytes).unwrap())
```

Comment:
I'd prefer not to unwrap here; since we're reading data from the user's file system it's very possible that it doesn't deserialize correctly. We should just return an error in this case (note that unwrapping when we serialize, in the previous function, should be fine since we've validated/constructed the data ourselves)